Source/GSXib5KeyedUnarchiver.m

```diff
@@ -0,0 +1,3391 @@
+                                                            andAttributes: [NSDictionary dictionaryWithObjectsAndKeys:
+                                                                                           @"NSMutableDictionary", @"class",
+                                                                                         refID, @"id",
+                                                                                         nil]];
+      GSXibElement *stringRecord   = [[GSXibElement alloc] initWithType: @"string"
+                                                            andAttributes: [NSDictionary dictionaryWithObjectsAndKeys:
+                                                                                           @"NS.key.0", @"key",
+                                                                                         nil]];
+      GSXibElement *placeholderRec = [[GSXibElement alloc] initWithType: @"object"
+                                                            andAttributes: [NSDictionary dictionaryWithObjectsAndKeys:
+                                                                                           @"IBUserDefinedRuntimeAttributesPlaceholder", @"class",
+                                                                                         @"IBUserDefinedRuntimeAttributesPlaceholderName", @"key",
+                                                                                         nil]];
+      GSXibElement *placeStringRec = [[GSXibElement alloc] initWithType: @"string"
+                                                            andAttributes: [NSDictionary dictionaryWithObjectsAndKeys:
+                                                                                           @"name", @"key",
+                                                                            nil]];
+      GSXibElement *referenceRec   = [self createReference: [element attributeForKey: @"id"]];
+
+      // Setup placeholder record...
+      [placeStringRec setValue: @"IBUserDefinedRuntimeAttributesPlaceholderName"];
+      [placeholderRec setElement: placeStringRec forKey: @"name"];
+      [placeholderRec setElement: referenceRec forKey: @"object"];
+      [placeholderRec setElement: element forKey: @"userDefinedRuntimeAttributes"];
+
+      // Attach placeholder and string records to the object record...
+      [objectRecord setElement: stringRecord forKey: @"NS.key.0"];
+      [objectRecord setElement: placeholderRec forKey: @"IBUserDefinedRuntimeAttributesPlaceholderName"];
+
+      // Add to flattened properties...
+      [_flattenedProperties setElement: objectRecord forKey: refID];
+
+      // Cleanup...
+      RELEASE(stringRecord);
+      RELEASE(placeholderRec);
+      RELEASE(placeStringRec);
+      RELEASE(objectRecord);
+    }
+}
+
+- (id) initForReadingWithData: (NSData*)data
+{
+  // Check data...
+  if (data == nil)
+    {
+      DESTROY(self);
+      return nil;
+    }
+#if     GNUSTEP_BASE_HAVE_LIBXML
```

Comment:
What if we have an `NSXML` implementation that doesn't have `libxml2`?

Reply:
The strange thing is that we actually have an XML parser that works without libxml2, so that part would work, but NSXMLDocument, which only gets used to decide whether we have the correct sort of XML, will only work correctly when that library is present. maybe I should make this check more local? But what would be the default if we don't have libxml2? Old XIB or new XIB?

Reply:
I moved that code over to GSXibLoader.m and only protect the NSDocument part.